cmd/cdi-controller/leaderelection.go

```diff
@@ -30,10 +33,19 @@ const (
 
 func startLeaderElection(ctx context.Context, config *rest.Config, leaderFunc func()) error {
 	client := kubernetes.NewForConfigOrDie(config)
+	cdiClient := cdiclient.NewForConfigOrDie(config)
 	namespace := util.GetNamespace()
 
 	// create manually so it has CDI component label
-	err := createConfigMap(client, namespace, configMapName)
+	crList, err := cdiClient.CdiV1beta1().CDIs().List(context.TODO(), metav1.ListOptions{})
```

Comment:
I'd like to keep our interactions with the CDI CR to a minimum.  Ideally just the operator.  I understand that is not always possible.  But I think it may be better to have the operator pass these label values as env vars to the deployments.  Then the values can just be cached locally.  If the values happen to change, the deployments will get updated.  THink about it

Reply:
That sounds better, just to be sure -> 
1. This means I'd have to start with `controller-lifecycle-operator-sdk`, make sure that `operator pass these label values as env vars to the deployments` is completed first?
(After this is in, I'll be able to fetch the env vars containing the values in the controllers)
2. Your review is tagging a specific file but I'm guessing the same goes for most occurrences like the ones in https://github.com/kubevirt/containerized-data-importer/pull/1864/files#diff-868af4b730094e9180f78af4cc069ae6cecf7077a46c186553f331dccbd32993 for example?